cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/BrokerMetricSample.java

```diff
@@ -103,7 +124,7 @@ public int brokerId() {
   public byte[] toBytes() {
     byte[] hostBytes = (entity().group() != null ? entity().group() : "UNKNOWN").getBytes(StandardCharsets.UTF_8);
     ByteBuffer buffer = ByteBuffer.allocate(457 + hostBytes.length);
-    buffer.put(CURRENT_VERSION);
+    buffer.put(_deserializationVersion);
```

Comment:
if it is `V4`, I think we do not need to write the `V5` metrics to byte stream, since they carry no information and  `readV4()` will ignore anyway.

Reply:
The reason why we serialize with the latest payload is to avoid maintaining multiple deprecated serialization versions. This cost of writing additional fields will only be relevant to the users having the deprecated broker serde versions. As a side-note, once/if we move towards a better serialization format such as flatbuffers, such nullable fields can implicitly be shrunk in size on the wire.